include/net/nrf_cloud.h

```diff
@@ -697,6 +697,59 @@ int nrf_cloud_init(const struct nrf_cloud_init_param *param);
  */
 int nrf_cloud_uninit(void);
 
+/**
+ * @brief Print details about cloud connection.
+ *
+ * if @kconfig{CONFIG_NRF_CLOUD_VERBOSE_DETAILS} is not enabled,
+ * only print the device id. If enabled, also print the protocol,
+ * sec tag, host name, stage, and team id.
+ *
+ * @return A negative value indicates an error.
+ */
+int nrf_cloud_print_details(void);
+
+/**
+ * @brief Retrieve the IMEI.
+ *
+ * If @kconfig{CONFIG_NRF_MODEM_LIB} is not enabled, returns an error.
+ *
+ * @param[in,out] buf A pointer in which to store the IMEI string.
```

Comment:
note that this buffer is used for the AT command response, which is larger than the IMEI.

if the user provides a 16 byte buffer for their IMEI string, i'm assuming `nrf_modem_at_cmd` will fail with `-NRF_E2BIG`?

this function, `nrf_cloud_get_uuid`, and `nrf_cloud_get_modem_fw` could probably just be moved into nrf_cloud library specific header since they are only used internally.  
unless you think they are valuable for users.

Reply:
Yes, it returns `-E2BIG`.